test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/PrimitiveTypeTests1.java

```diff
@@ -53,6 +55,21 @@ public class PrimitiveTypeTests1 {
 	private static final SymbolLookup nativeLibLookup = SymbolLookup.loaderLookup();
 	private static final SymbolLookup defaultLibLookup = linker.defaultLookup();
 
+	static MemorySegment allocateString(String string) throws Throwable {
+		MemorySegment segment;
+
+		if (isZos) {
+			byte[] bytes = string.getBytes("IBM1047");
+			byte[] withNul = Arrays.copyOf(bytes, bytes.length + 1);
+
+			segment = arena.allocateFrom(JAVA_BYTE, withNul);
```

Comment:
This can be simplified:
```java
			segment = arena.allocateFrom(string, Charset.forName("IBM1047"));
```

Reply:
We can't do this according to https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/foreign/SegmentAllocator.html#allocateFrom(java.lang.String,java.nio.charset.Charset)
and https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/nio/charset/Charset.html & https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/nio/charset/StandardCharsets.html,
in which the 2r argument of `allocateFrom` is `Charset` rather then Charset name (`IBM1047` is not defined in `Charset`).
That's why we have to convert via `string.getBytes(charset name)`.

Reply:
According to https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/nio/charset/StandardCharsets.html & https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/nio/charset/Charset.html:
```
static final Charset
ISO_8859_1  ISO Latin Alphabet No. 1, also known as ISO-LATIN-1.
static final Charset
US_ASCII  Seven-bit ASCII, also known as ISO646-US, also known as the Basic Latin block of the Unicode character set.
static final Charset
UTF_16  Sixteen-bit UCS Transformation Format, byte order identified by an optional byte-order mark.
static final Charset
UTF_16BE  Sixteen-bit UCS Transformation Format, big-endian byte order.
static final Charset
UTF_16LE   Sixteen-bit UCS Transformation Format, little-endian byte order.
static final Charset
UTF_32  Thirty-two-bit UCS Transformation Format, byte order identified by an optional byte-order mark.
static final Charset
UTF_32BE   Thirty-two-bit UCS Transformation Format, big-endian byte order.
static final Charset
UTF_32LE  Thirty-two-bit UCS Transformation Format, little-endian byte order.
static final Charset
UTF_8  Eight-bit UCS Transformation Format.
```
I suspect "IBM1047" is not supported in Charset (this way we have to reply on `string.getBytes("IBM1047")` for conversion) and there is no way to verify this case on z/OS for now.

Reply:
Or we can have this changed for now but it needs to be verified on JDK22+ on z/OS

Reply:
IBM1047 isn't a "standard" charset (that is it isn't exposed as a field of `StandardCharsets`), but it will be available, especially on z/OS; see https://github.com/ibmruntimes/openj9-openjdk-jdk22/blob/openj9/make/data/charsetmapping/IBM1047.map.

`String.getBytes("IBM1047")` also relies on that fact (it's effectively equivalent to `String.getBytes(Charset.forName("IBM1047"))`.

Reply:
Fixed with `Charset.forName` as suggested at https://github.com/eclipse-openj9/openj9/pull/19689#discussion_r1639880750 given `IBM1047` is supported on z/OS.